fastlane/fastlane
# [pilot] improves testflight metadata upload

### Checklist

### Motivation and Context

The current `upload_to_testflight` action allows users to submit a `changelog` to AppStoreConnect once the build has completed processing.

It turns out that **full** processing is actually not required for the changelog to be submitted. Once a build **appears** in AppStoreConnect, we are able to send the changelog information upstream without waiting for the full processing of the build to occur.

The goal of this PR is to introduce a slight change in behavior to `upload_to_testflight`, where if a user specifically sets `skip_waiting_for_build_processing` to `true` but also includes a `changelog`, we honor the changelog request by only waiting until the build appears on AppStoreConnect, and then bailing (the current behavior is to simply ignore the changelog, which is often un-intuitive and most likely undesirable).

### Description

This PR changes the internal of `build_manager` and `build_watcher` slightly.

On `build_manager` we determine whether the `skip_waiting_for_build_processing` was set to `true` and `changelog` is not `nil`.

Under this condition, we pass a new `skip_full_processing` variable to the `build_watcher` that allows us to poll the build on AppStoreConnect and bypass the most time consuming part of the processing which happens after the build appears on AppStoreConnect.

Note that this change is an implementation detail that hasn't had any effect on existing tests. I'd be more than happy to add a new test for this scenario but based on the existing tests it felt unnecessary and perhaps redundant but I'm happy to be convinced otherwise.

### Testing Steps

Run `fastlane run testflight skip_waiting_for_build_processing:true changelog:"test"` and ensure that once an IPA is uploaded, that the polling of the build only waits for the build to appear on AppStoreConnect and successfuly exist the process.
